### PR TITLE
[Core] Remove thread local core worker instance 2/n

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
We removed the thread local core worker instance in this PR, which is the further arch cleaning stuff for removing multiple workers in one process.
It also removes the unnecessary parameter `workerId` from JNI.

## Related issue number
#24567

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Lmk when it is ready to review!

**Comment by jovany-wang** [comment]
@kfstorm @rkooo567 It's ready for review now.

File: src/ray/core_worker/core_worker_process.cc [hunk 1]
```diff
@@ -151,6 +147,13 @@ CoreWorkerProcessImpl::~CoreWorkerProcessImpl() {
   if (options_.enable_logging) {
     RayLog::ShutDownRayLog();
   }
+
+  {
+    absl::WriterMutexLock lock(&mutex_);
+    if (core_worker_ != nullptr) {
+      core_worker_ = nullptr;
```
Comment (kfstorm) [inline src/ray/core_worker/core_worker_process.cc:154]: Why add this section? The reset of `core_worker_` is already done in `ShutdownDriver` and `RunWorkerTaskExecutionLoop`.

**Review by kfstorm — APPROVED** [review]
Nice work! one small comment.

File: src/ray/core_worker/core_worker_process.cc (hunk 1 above)
Comment (jovany-wang) [reply src/ray/core_worker/core_worker_process.cc:154]: Good catch.

**Comment by jovany-wang** [comment]
Failures are not related.

**Comment by scv119** [comment]
huge improvement!

**Pull request merged**